lib/fluent/supervisor.rb

```diff
@@ -563,12 +563,8 @@ def init(process_type, worker_id)
           if @log_rotate_age || @log_rotate_size
             # We need to prepare a unique path for each worker since
             # Windows locks files.
-            if Fluent.windows?
-                path = LoggerInitializer.per_process_path(@path, process_type, worker_id)
-            else
-                path = @path
-            end
-            @logdev = Fluent::LogDeviceIO.new(path, shift_age: @log_rotate_age, shift_size: @log_rotate_size)
+            @path = LoggerInitializer.per_process_path(@path, process_type, worker_id) if Fluent.windows?
```

Comment:
`@path` should be set at the head of this method.

Reply:
Yes, can I fix it in the next PR that I'm planning to fix #4037?

I plan to fix it so that the path is passed from outside `init()`, in order to avoid logging inside the `LoggerInitializer`.

Reply:
I see, please go ahead.